src/hooks/useActiveDescendantListFocus/index.js

```diff
@@ -143,47 +42,72 @@ function useActiveDescendantListFocus({
       stopPropagation: true
     };
   }, [useDocumentEventListeners, focusedElementRef]);
-
-  useKeyEvent({
-    keys: [nextArrow],
-    callback: onArrowNext,
-    ...listenerOptions
+  const { triggerByKeyboard, setTriggerByKeyboard } = useIsTriggerByKeyboard({ focusedElementRef });
+  const setVisualFocusItemId = useCallback(
+    (visualFocusItemId, isTriggeredByKeyboard) => {
+      if (!triggerByKeyboard) setTriggerByKeyboard(isTriggeredByKeyboard);
+      const itemIndex = itemsIds.indexOf(visualFocusItemId);
+      if (itemIndex > -1 && itemIndex !== visualFocusItemIndex) {
+        setVisualFocusItemIndex(itemIndex);
+      }
+    },
+    [itemsIds, setTriggerByKeyboard, triggerByKeyboard, visualFocusItemIndex]
+  );
+  useSupportArrowsKeyboardNavigation({
+    itemsCount,
+    focusedElementRef,
+    visualFocusItemIndex,
+    setVisualFocusItemIndex,
+    triggerByKeyboard,
+    setTriggerByKeyboard,
+    isHorizontalList,
+    isItemSelectable,
+    listenerOptions
   });
 
-  useKeyEvent({
-    keys: [backArrow],
-    callback: onArrowBack,
-    ...listenerOptions
+  useSupportPressItemKeyboardNavigation({
+    visualFocusItemIndex,
+    itemsCount,
+    focusedElementRef,
+    setVisualFocusItemIndex,
+    onItemClick,
+    isItemSelectable,
+    listenerOptions,
+    isIgnoreSpaceAsItemSelection
   });
 
-  useKeyEvent({
-    keys: pressKeys,
-    callback: keyboardOnSelectCallback,
-    ...listenerOptions
+  SetDefaultItemOnFocusEvent({
+    focusedElementRef,
+    isItemSelectable,
+    visualFocusItemIndex,
+    setVisualFocusItemIndex,
+    itemsCount,
+    triggerByKeyboard,
+    defaultVisualFocusItemIndex
   });
 
-  useEventListener({
-    eventName: "blur",
-    ref: focusedElementRef,
-    callback: onBlurCallback
+  useKeepFocusOnItemWhenListChanged({
+    visualFocusItemIndex,
+    itemsIds,
+    isItemSelectable,
+    setVisualFocusItemIndex
   });
 
-  // if element unmount act like element got blur event
-  useEffect(() => {
-    // if element unmount
-    if (focusedElementRef?.current === null && previousFocusedElementRef?.current !== null) {
-      onBlurCallback();
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [focusedElementRef.current, previousFocusedElementRef, onBlurCallback]);
+  // this callback function is not needed anymore (the developer not need to replace is element on click with this callback)
```

Comment:
```suggestion
  // this callback function is not needed anymore (the developer does not need to replace  the element's on click with this callback)  - we keep it for backward compatibility
```

Reply:
remove second comment and change variable name to include backward compatibility